core/src/test/scala/unit/kafka/admin/ReassignPartitionsUnitTest.scala

```diff
@@ -641,4 +642,21 @@ class ReassignPartitionsUnitTest {
     assertTrue("Expected the string to start with %s, but it was %s".format(prefix, str),
       str.startsWith(prefix))
   }
+
+  @Test
+  def testPropagateInvalidJsonError(): Unit = {
+    val adminClient = new MockAdminClient.Builder().numBrokers(4).build()
+    try {
+      addTopics(adminClient)
+      assertStartsWith("Unexpected character",
+        assertThrows(
+          classOf[JsonParseException], new ThrowingRunnable {
```

Comment:
Why do we need a `ThrowingRunnable` here?

Reply:
Good point, refactored it a bit